src/codeu/chat/server/Controller.java

```diff
@@ -22,20 +22,37 @@
 import codeu.chat.common.RandomUuidGenerator;
 import codeu.chat.common.RawController;
 import codeu.chat.common.User;
+import codeu.chat.database.Database;
 import codeu.chat.util.Logger;
 import codeu.chat.util.Time;
 import codeu.chat.util.Uuid;
+import com.mongodb.BasicDBObject;
+import com.mongodb.DBCursor;
+import com.mongodb.client.FindIterable;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoCursor;
+import org.bson.BsonDocument;
+import org.bson.Document;
+import org.bson.codecs.configuration.CodecRegistry;
+import org.bson.conversions.Bson;
 
 public final class Controller implements RawController, BasicController {
 
   private static final Logger.Log LOG = Logger.newLog(Controller.class);
 
   private final Model model;
+  private final Database database;
+
   private final Uuid.Generator uuidGenerator;
+  private final Uuid serverId;
 
-  public Controller(Uuid serverId, Model model) {
+  public Controller(Uuid serverId, Model model, Database database) {
     this.model = model;
     this.uuidGenerator = new RandomUuidGenerator(serverId, System.currentTimeMillis());
+    this.serverId = serverId;
+    this.database = database;
+
+    loadDatabase();
```

Comment:
This is the comment that is most confusing to me. I understand why not to do "real work" within the constructor, but I don't know how else to do it. You mention a static factory, but I guess I'm not fully aware of what that is.